### PR TITLE
stream_info: Make some struct members private

### PR DESCRIPTION
**Pull request by abeyad**

### CODE REVIEW

**Comment by kyessenov** [comment]
Why close? I think this is a good fix.

**Comment by abeyad** [comment]
> Why close? I think this is a good fix.

Hey @kyessenov thanks!  I ran into some issues w/ the PR and wasn't sure when I would be able to get back to it, so closed it, but I just tried a fix so re-opening the PR now.

**Comment by ravenblackx** [comment]
@kyessenov ping

File: source/common/stream_info/stream_info_impl.h [hunk 1]
```diff
@@ -437,26 +437,22 @@ struct StreamInfoImpl : public StreamInfo {
...
-public:
-  absl::optional<std::string> response_code_details_;
-  absl::optional<std::string> connection_termination_details_;
   absl::InlinedVector<ResponseFlag, 4> response_flags_{};
   bool health_check_request_{};
   Router::RouteConstSharedPtr route_;
   envoy::config::core::v3::Metadata metadata_{};
   FilterStateSharedPtr filter_state_;
+
+private:
```
Comment (kyessenov) [inline source/common/stream_info/stream_info_impl.h:447]: Can we keep the order of the fields the same? This is a fairly common data structure and the layout would change if we-order - possibly for worse.

Comment (abeyad) [reply source/common/stream_info/stream_info_impl.h:454]: The tests have a size check which passed: https://github.com/envoyproxy/envoy/blob/c6b8553313b818e981cad06d0d2a9d190df92c29/test/common/stream_info/stream_info_impl_test.cc#L44-L48, so I assumed the ordering was fine, but given they check a bunch of different sizes, I reverted back to the same ordering as before.  PTAL.

**Review by kyessenov — APPROVED** [review]

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**